napari_plugin_manager/base_qt_plugin_dialog.py

```diff
@@ -0,0 +1,1485 @@
+import contextlib
+import os
+import webbrowser
+from functools import partial
+from typing import (
+    Dict,
+    List,
+    Literal,
+    NamedTuple,
+    Optional,
+    Protocol,
+    Sequence,
+    Tuple,
+)
+
+from packaging.version import parse as parse_version
+from qtpy.QtCore import QSize, Qt, QTimer, Signal, Slot
+from qtpy.QtGui import (
+    QAction,
+    QActionGroup,
+    QFont,
+    QKeySequence,
+    QShortcut,
+)
+from qtpy.QtWidgets import (
+    QCheckBox,
+    QComboBox,
+    QDialog,
+    QFrame,
+    QGridLayout,
+    QHBoxLayout,
+    QLabel,
+    QLineEdit,
+    QListWidget,
+    QListWidgetItem,
+    QMenu,
+    QPushButton,
+    QSizePolicy,
+    QSplitter,
+    QTextEdit,
+    QToolButton,
+    QVBoxLayout,
+    QWidget,
+)
+from superqt import QCollapsible, QElidingLabel
+
+from napari_plugin_manager.qt_package_installer import (
+    InstallerActions,
+    InstallerQueue,
+    InstallerTools,
+    ProcessFinishedData,
+)
+from napari_plugin_manager.qt_widgets import ClickableLabel
+from napari_plugin_manager.utils import is_conda_package
+
+# Scaling factor for each list widget item when expanding.
+CONDA = 'Conda'
+PYPI = 'PyPI'
+
+
+class BasePackageMetadata(Protocol):
```

Comment:
Checking, I think this is needed as it is so it can be used for the Qt `Slot` decorator (putting this definition under a `TYPE_CHECKING` guard raises an error - `NameError: name 'BasePackageMetadata' is not defined` when using `Slot`). However, if there is a better way to handle this let me know

Reply:
If that's the case let's add a comment about in the docstring, thanks! I'm not aware of any other alternatives.